examples/runc-hook/main.go

```diff
@@ -0,0 +1,222 @@
+	kubeconfig             = flag.String("kubeconfig", "", "kubeconfig")
+	node                   = flag.String("node", "", "Node name")
+
+	notifier *runcfanotify.RuncNotifier
+	client   *kubernetes.Clientset
+)
+
+func publishEvent(reason, message string) {
+	eventTime := metav1.NewTime(time.Now())
+	event := &api.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%v.%x", *node, time.Now().UnixNano()),
+			Namespace: "default",
+		},
+		Source: api.EventSource{
+			Component: "RuncHook",
+			Host:      *node,
+		},
+		Count:               1,
+		ReportingController: "github.com/kinvolk/inspektor-gadget",
+		ReportingInstance:   os.Getenv("POD_NAME"), // pod name
+		FirstTimestamp:      eventTime,
+		LastTimestamp:       eventTime,
+		InvolvedObject: api.ObjectReference{
+			Kind: "Node",
+			Name: *node,
+			// Uncomment to make it visible in 'kubectl describe node'
+			//UID: types.UID(*node),
+		},
+		Type:    api.EventTypeNormal,
+		Reason:  reason,
+		Message: message,
+	}
+
+	if _, err := client.CoreV1().Events("default").Create(context.TODO(), event, metav1.CreateOptions{}); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to create event: %s\n", err)
+	}
+}
+
+func callback(notif runcfanotify.ContainerEvent) {
+	// The OCI State is defined at:
+	//     https://github.com/opencontainers/runtime-spec/blob/main/runtime.md#state
+	// and is passed to OCI hooks over stdin, see:
+	//     https://github.com/opencontainers/runtime-spec/blob/main/config.md#posix-platform-hooks
+	// ociState will be given as stdin to the command.
+	//
+	// This code does not add any hooks in the config but executes the
+	// command in the same way that OCI hooks would do. For this we need to
+	// synthesise the OCI State.
+	ociState := &ocispec.State{
+		Version: ocispec.Version,
+		ID:      notif.ContainerID,
+		Pid:     int(notif.ContainerPID),
+		// TODO: Make runcfanotify return the bundle dir path too.
+		Bundle: "",
+	}
+	if notif.ContainerConfig != nil && notif.ContainerConfig.Annotations != nil {
+		ociState.Annotations = notif.ContainerConfig.Annotations
+	} else {
+		ociState.Annotations = make(map[string]string)
+	}
+
+	var cmd string
+	switch notif.Type {
+	case runcfanotify.EventTypeAddContainer:
+		ociState.Status = ocispec.StateCreated
+		var config string
+		if notif.ContainerConfig != nil {
+			b, err := json.MarshalIndent(notif.ContainerConfig, "", "  ")
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to marshall ContainerConfig: %s\n", err)
+			} else {
+				config = string(b)
+			}
+		}
+		if outputAdd {
+			fmt.Printf("Container added: %v pid %d\n", notif.ContainerID, notif.ContainerPID)
+		}
+		if outputConfig {
+			if config != "" {
+				fmt.Printf("%s\n", config)
+			} else {
+				fmt.Fprintf(os.Stderr, "Error: container config not found for container %s\n", notif.ContainerID)
+			}
+		}
+		if *publishKubernetesEvent {
+			publishEvent("NewContainerConfig", config)
```

Comment:
heh, if the config is empty the event is rather poor. But seems fine for an early version :)